src/main/scala/outwatch/Sink.scala

```diff
@@ -48,35 +50,39 @@ sealed trait Sink[-T] extends Any {
 
 object Sink {
 
-  // these classes need to be in this file, because Sink is sealed
-  private[outwatch] case class ObservableSink[-I, +O](
-    sink: Sink[I], source: Observable[O]
-  ) extends Observable[O](source.inner) with Sink[I]{
-    override private[outwatch] def observer = sink.observer
+  private[outwatch] final case class ObservableSink[-I, +O](oldSink: Sink[I], stream: Observable[O]) extends Observable[O] with Sink[I] {
+    override private[outwatch] def observer = oldSink.observer
+
+    override def unsafeSubscribeFn(subscriber: Subscriber[O]): Cancelable = stream.unsafeSubscribeFn(subscriber)
   }
 
-  private[outwatch] final case class SubjectSink[T]() extends Subject[T](new SubjectFacade) with Sink[T] {
-    override private[outwatch] def observer = this
+  private[outwatch] final case class SubjectSink[T]()(implicit scheduler: Scheduler) extends Observable[T] with Sink[T] {
```

Comment:
Why exactly do we need the scheduler here? Could we not add it to the methods? This would unclutter our code a bit, no?

Reply:
In fact, why don't we do:

`def observer: Observer[T]`? So we wouldn't need the scheduler at all?

Reply:
We need it to create the `Subscriber` inside the `SubjectSink` (the `Sink.observer` is actually a `Subscriber` now) . If I would have left it an `Observer`, all the `Sink.redirect` operations would have required a `Scheduler` (and also many operations in `PipeOps`). This way they just use the `Scheduler` from the source `Sink`'s `Subscriber`.

Reply:
Yeah, that makes sense, however, I think creating a new `Handler` or `Pipe` is going to be a much much more common operation, then `redirect`, so I think it'd be better if the individual methods required the implicit evidence, no?

Reply:
Not quite, `EmitterBuilder`'s `-->` would require an implicit `Scheduler` in that case.